feat(cli): modular `pattern` namespace (#319)
ACCIDDA/flepimop2: docs/index.md

## Create a Project

> Scaffold a new project with `flepimop2 pattern`, then move into it:

Reviewer: ```suggestion
You can create a new project with `flepimop2 pattern <target_project_directory>`
```

aside: how does this behave if `quickstart-project` directory already exists?

```bash
flepimop2 pattern quickstart-project
cd quickstart-project
```

This creates the standard project structure:

> ```

Reviewer: maybe change this to have the cd directory + ls -R command to view in the second block?

├── model_input/
│   ├── data/
│   └── plugins/
│       ├── SIR.py
│       └── solve_ivp.py
├── model_output/
├── postprocessing/
│   └── SIR_plot.R
└── README.md
```

> Every flepimop2 project needs at least three things to run: a configuration file, a system, and an engine. The **configuration file** (saved in `configs`) is a YAML file that specifies your model parameters, which system and engine to use, where to write outputs, and optionally what post-processing steps to run after a simulation. The **system** and **engine** are backends that implement the model dynamics and the numerical solver, respectively. In this quickstart, we will use Python scripts (saved in `model_input/plugins`) for both the system and the engine. `flepimop2 pattern` already places those files in the correct locations, includes the dependencies required for this page, and includes the post-processing script used later in the guide.

Reviewer: I think its important to state early on what's going on with defaults here:

```
$ flepimop2 pattern <target_project>
```

is *actually*

```
$ flepimop2 pattern --module copy --source <flepimop2-docs-example> <target_project>
```

or some such. After getting familiar with the tool, people are likely going to want to be sourcing off other projects generally.

Don't know if that explanation appears roughly where I've made this note or elsewhere.
